### PR TITLE
Update RVI SOTA Client

### PR DESCRIPTION
**Pull request by leon-anavi**
Update RVI SOTA Client to the version used in Automotive Grade Linux (AGL). The patches include recies for libgit2, libssh2, lshw, new version of rvi-sota-client and new version of meta-rust.

Best regards,
Leon

### CODE REVIEW

**Comment by zeenix** [comment]
"libgit2: Required by RVI SOTA Client"

The shortlog should summarize the change rather than rationale (which is for description part). Same goes for libssh2 patch.

**Comment by leon-anavi** [comment]
@zeenix, thanks. I have updated the commit messages.

Best regards, Leon

**Comment by zeenix** [comment]
It builds fine here and I trust you when you say that you have tested the changes. Before I can merge, I think the commit logs can still be better:

"libssh2: C library with SSH2 implementation" doesn't summarize the change at all but rather the component being changed. Moroever the description is also vauge since it says who needs the change but no information on change itself or why SOTA needs it exactly.

"libgit2: The Git linkable library" this has the same issue. "Add libgit2" would be fine here and you can describe the library in the description. Same goes for "lshw: Hardware lister" shortlog.

"rvi-sota-client: Update version" is on the very right track. :) However, let's also specify what it's been updated to. Preferrably in the shortlog itself but if not, then in description (since shortlog is supposed to be ideally no more than 50 chars long).

**Comment by leon-anavi** [comment]
Hi @zeenix,

I have updated the git commit message following your recommendations. Now I hope that they meet your requirements and my patches will be merged. Please kindly describe the requirements for the git commit messages and provide examples in GENIVI wiki. This will make life easier for both new and existing contributors.

Best regards,
Leon

**Comment by zeenix** [comment]
Thanks. As I said on IRC, these are not my personal requirements but rules that are followed in all git projects that stress on quality when it comes to code history. I can create a wikipage but it'll have to be just a link to https://wiki.gnome.org/Git/CommitMessages.

**Comment by zeenix** [comment]
I think you missed the last comment:

"rvi-sota-client: Update version" is on the very right track. :) However, let's also specify what it's been updated to. Preferrably in the shortlog itself but if not, then in description (since shortlog is supposed to be ideally no more than 50 chars long).

**Comment by leon-anavi** [comment]
Any format of the commit messages is fine for me if there are exact rules and examples. 

@chbae and @tom--pollard are adding the id and the title of the JIRA issue at the bottom of the commit message. If you want we can keep this way or switch to the format as in GNOME with URL to the issues. It is up to you. For me it is just important to have the exact requirements well described and available in the wiki or another public URL.

Thanks, Leon

File: meta-genivi-dev/recipes-support/lshw/files/cross-compile.patch [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ 3 files changed, 4 insertions(+), 4 deletions(-)
+
+--- a/src/Makefile
++++ b/src/Makefile
+@@ -18,7 +18,7 @@ export MANDIR
+ export DATADIR
+ export SQLITE
+ 
+-CXX?=c++
++CXX?=$(CROSS_COMPILE)c++
```
Comment (zeenix) [inline meta-genivi-dev/recipes-support/lshw/files/cross-compile.patch:14]: These changes are not part of this logical change (addition of lshw) even if they are required by lshw recipe. It's OK for now but for future reference, each logical change goes into it's own patch with it's own description and rationale.

**Comment by tom--pollard** [comment]
The JIRA ticket tag & title is required for the JIRA - Github integration, allowing the status of the PR to be visible from within the JIRA ticket.

**Comment by zeenix** [comment]
Ah sorry I didn't think of Jira references. Yeah, we should keep the existing practice. So I guess we need our own wikipage after all. :)

**Comment by zeenix** [comment]
""Update the version of RVI SOTA Client to the one
used in Automotive Grade Linux (AGL)"

Is that latest git master or is there a tag for that version? If so, we need to specify that here. Otherwise, this is fine.

**Comment by leon-anavi** [comment]
Yocto/OE recipe rvi-sota-client_git.bb contains the exact version in SRCREV. I don't have permissions to create tags in repository advancedtelematic/rvi_sota_client. I can kidly ask ATS Advanced Telematic Systems to tag this version.

Best regards, Leon

**Comment by zeenix** [comment]
No need really. I just wanted to be more specific than "same as XYZ project uses" but commit hash doesn't really add any meta-info to commit log and if anyone wants the specifics, they can look at the code changes themselves.

**Comment by leon-anavi** [comment]
The git log of advancedtelematic/rvi_sota_client contains quite a lot of specific therefore in my opion the commit hash is a useful milestone. Of course you are right that if anyone wants more details than just gi logs, they can have look at the code changes themselves.

Anyway, thank you for merging my changes. Best regards, Leon

**Pull request merged**